packages/text-field/src/TextField/TextField.tsx

```diff
@@ -0,0 +1,135 @@
+  variant?: "outline" | "ghost";
+  disabled?: boolean;
+  error?: boolean | string;
+  value?: string;
+  onChange?: (
+    value: string,
+    { event }: { event: React.ChangeEvent<HTMLInputElement> },
+  ) => void;
+  leadingIcon?: React.ComponentProps<typeof Icon>;
+  TrailingAction?: React.ReactElement;
+};
+
+type TextFieldRef = HTMLInputElement;
+
+type InternalProps = {
+  id: string;
+  state: "default" | "error" | "disabled";
+  size: TextFieldProps["size"];
+  variant: TextFieldProps["variant"];
+  message?: string;
+};
+
+const TextField = React.forwardRef<TextFieldRef, TextFieldProps>(
+  (
+    {
+      size = "2",
+      variant = "outline",
+      value,
+      onChange,
+      leadingIcon,
+      TrailingAction,
+      className,
+      disabled,
+      error,
+      ...props
+    },
+    forwardedRef,
+  ) => {
+    const inputRef = React.useRef<HTMLInputElement>(null);
+    const composedRefs = useComposedRefs(inputRef, forwardedRef);
+
+    const [internalState, setInternalState] =
+      React.useState<InternalProps["state"]>("default");
+
+    React.useEffect(() => {
+      if (disabled) {
+        setInternalState("disabled");
+      } else if (error) {
+        setInternalState("error");
+      } else {
+        setInternalState("default");
+      }
+    }, [disabled, error]);
+
+    // Adds default props to the component, which can be overridden by the caller
+    const TrailingActionComponent = TrailingAction
+      ? React.cloneElement(TrailingAction, {
+          size,
+          color: "gray",
+          variant: "ghost",
+          ...TrailingAction?.props,
+        })
+      : null;
+    return (
+      <div
+        className={clsx(
+          "box-border flex items-center transition-all",
+          "border-2 border-solid text-gray-12 placeholder:text-gray-10",
+          COLOR.Container[internalState][variant],
+          SIZE.Container[size],
+        )}
+        // Focus the input when clicking on the container
+        onPointerDown={(event) => {
+          const target = event.target as HTMLElement;
+
+          // Make sure we're not clicking on an interactive element
+          if (target.closest("button, a")) {
+            event.preventDefault();
+            return;
+          }
+
+          const input = inputRef.current;
+          if (!input) return;
+
+          requestAnimationFrame(() => {
+            input.focus();
+          });
+        }}
+      >
+        {leadingIcon && <Icon size={size} color="gray" {...leadingIcon} />}
+        <input
+          value={value}
+          onChange={(event) => {
+            onChange?.(event.target.value, { event });
+          }}
+          className={clsx(
+            "appearance-none text-gray-12 border-none shadow-0 outline-0 bg-transparent",
+            "[font-family:inherit] h-full w-full",
+            SIZE.Field[size],
+            className,
+          )}
+          disabled={disabled}
+          {...props}
+          ref={composedRefs}
```

Comment:
I'm assuming this component is meant to be <input type="text" /> given the name. If so should the type be set explicitly? If not, should this be a generic `Input` component then?

Reply:
I'm not understanding

Reply:
I didn't add `type="text"` to this because things like `password` and `email` exist which would work fine here. I could narrow the type prop to only accept those types?

Reply:
Maybe the move is to just make this an `Input` component then, it seems that's what it is. I know that is not a thing in the figma file, but then again the implmementation and the figma design system should evolve together to match.